### ISSUE TITLE
"Save the game before sleeping" needs slight adjustments

### ISSUE THREAD

**Issue by G3Kappa** [CleverRaven/Cataclysm-DDA#25222]
This is a very minor feature request. When trying to sleep, the game offers a number of options and one of them is to save the game. There are two problems with the current implementation, in my opinion:

1. The game is saved even if the player character does not fall asleep. To me, it would make more sense if the game was saved _only if the attempt at falling asleep succeeds_, so that you don't have to save a bunch of times for no reason. If you still want to save before actually going to sleep, you can always do a quick save.

2. If you have an alarm, you can't save before sleeping while also setting your desired wake-up time ([see picture](https://i.imgur.com/xIZmdAU.png)). This could be solved easily by re-implementing the save function as a Y/N option.

I'd like to see the UI panel changed to look like this:

|  `KEY` | Are you sure you want to sleep?| 
| - | ------------- |
| `y` | Yes. | 
| `n` | No | 
| `<N>` | Set the alarm to wake up in \<N\> hours |
| `s` | And **[don't save/save]** the game before **falling asleep**. |

To recap, saving the game should be a Y/N option and the game should only be saved if the dice roll to fall asleep succeeds.

Alternatively, the alarm system could be decoupled and given its own UI. I don't know whether wrist watches, alarm clocks and internal chronometers differ internally, but if they don't it should be feasible if not a bit more involved.

**Comment by Leland** [comment]
> The game is saved even if the player character does not fall asleep

IIRC this was changed intentionally. game.cpp is too big to blame on GitHub, will need to open this up on the CLI to find the reasoning.

> If you have an alarm, you can't save before sleeping while also setting your desired wake-up time

This is a good point.

We could make sleeping a 2 step process – first panel is "Are you sure you want to sleep?" <kbd>y</kbd>/<kbd>n</kbd>/<kbd>s</kbd>, and if either <kbd>y</kbd>es or <kbd>s</kbd>ave are chosen, the second panel is "You have an alarm clock. Set an alarm?"  <kbd>n</kbd>/<kbd>3</kbd>/<kbd>5</kbd>, etc.

**Comment by G3Kappa** [comment]
> IIRC this was changed intentionally. game.cpp is too big to blame on GitHub, will need to open this up on the CLI to find the reasoning.

That's fine, the reason I opened this issue was the alarm thing actually. I figured I'd point it out while I was at it, but if it's intentional then it's fine.

Having read the relevant code, I can see how my initial proposal of including a toggled option would be less straightforward than your proposed solution, so I'm completely on board with what you said.
